Foundation/include/Poco/Dynamic/VarHolder.h

```diff
@@ -787,6 +823,287 @@ class VarHolderImpl: public VarHolder
 		return typeid(T);
 	}
 
+	void convert(Int8& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToSigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(Int16& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToSigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(Int32& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToSigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(Int64& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToSigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(UInt8& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToUnsigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(UInt16& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToUnsigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(UInt32& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToUnsigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(UInt64& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToUnsigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+#ifdef POCO_INT64_IS_LONG
+
+	void convert(long long& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToSigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(unsigned long long& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			convertToUnsigned(std::underlying_type_t<T>(_val), val);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+#endif
+
+	void convert(bool& val) const
+	{
+		if constexpr (std::is_enum_v<T>)
+		{
+			val = (std::underlying_type_t<T>(_val) != 0);
+		}
+		else
+		{
+			VarHolder::convert(val);
+		}
+	}
+
+	void convert(float& val) const
```

Comment:
Is conversion of enum to float meaningful?

Reply:
probably not, but I don't think it hurts